plugins/http_plugin/http_plugin.cpp

```diff
@@ -269,8 +270,16 @@ namespace eosio {
 
          if( options.count( "unix-socket-path" ) && !options.at( "unix-socket-path" ).as<string>().empty()) {
             boost::filesystem::path sock_path = options.at("unix-socket-path").as<string>();
-            if (sock_path.is_relative())
+            if (sock_path.is_relative()) {
                sock_path = app().data_dir() / sock_path;
+            }
+            // The maximum length of the socket path is defined by sockaddr_un::sun_path. On Linux,
+            // according to unix(7), it is 108 bytes. On FreeBSD, according to unix(4), it is 104 bytes.
+            const auto unix_socket_limits = sizeof(sockaddr_un::sun_path);
+            if (sock_path.string().size() >= unix_socket_limits) {
+               // try a relative path which may be shorter.
+               sock_path = boost::filesystem::relative(sock_path);
```

Comment:
If the relative path is still not short enough, should we `EOS_ASSERT`? Or do you get a decent error message without the `EOS_ASSERT`?

Reply:
A system exception would be thrown with 'path is too long' message.